services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -212,9 +216,62 @@ def engine(self) -> Engine:
             async for row in conn.execute(query):
                 result_dict = dict(row.items())
                 log.debug("found project: %s", result_dict)
+                tags = await self._get_tags_by_study(study_id=result_dict['id'])
+                result_dict['tags'] = tags
                 projects_list.append(_convert_to_schema_names(result_dict))
         return projects_list
 
+    async def _get_study(self, user_id: str, project_uuid: str, exclude_foreign: List = []) -> Dict: # pylint: disable=dangerous-default-value
+        async with self.engine.acquire() as conn:
+            joint_table = user_to_projects.join(projects)
+            query = select([projects]).select_from(joint_table).where(
+                and_(projects.c.uuid == project_uuid,
+                     user_to_projects.c.user_id == user_id)
+            )
+            result = await conn.execute(query)
+            study_row = await result.first()
+
+            if not study_row:
+                raise ProjectNotFoundError(project_uuid)
+
+            study = dict(study_row.items())
+
+            if 'tags' not in exclude_foreign:
+                tags = await self._get_tags_by_study(study_id=study_row.id)
+                study['tags'] = tags
+
+            return study
```

Comment:
I'm a bit confused by this. what is the difference now between a study and a project?
can we once define this? 
how does this changes with regard to the function get_user_project ? (it looks pretty similar)

Reply:
We agreed that a study is a project (entry in the db) and it data (stored in s3). Therefore here we should speak ONLY of projects

Reply:
The only difference is that the old one passes the results through that ``_convert_to_schema_names`` that hides some columns from the result. One of those columns is the study's id, which I need to get the tags.. (remember that I pointed out to the problems of this added complexity of having several ids?). I need the actual primary_key to get the tags. And as we are using the metadata approach and not the database classes approach to create out models (because of this compatibility problem with libraries), we cannot automatize the fetching of linked entities.
Anyhow I need the id, and I need the study to be a dict before I can add the tags to the result. ``_get_study`` is used only from inside that class and I tried to use it from the old one and I got a gazillion tests failing.
I will try again.